app/workers/updates_synchronizer_worker.rb

```diff
@@ -13,13 +13,15 @@ def perform(check_for_todays_file = true, reapply_data_migrations = false)
     if TradeTariffBackend.uk?
       TariffSynchronizer.download_cds
 
-      if check_for_todays_file &&
-          still_time_to_reschedule? &&
-          todays_file_has_not_yet_arrived?
-
-        self.class.perform_in(TRY_AGAIN_IN, true)
-        logger.info "Daily file missing, retrying at #{TRY_AGAIN_IN.from_now}"
-        return
+      if check_for_todays_file && todays_file_has_not_yet_arrived?
+        if still_time_to_reschedule?
+          self.class.perform_in(TRY_AGAIN_IN, true)
+          logger.info "Daily file missing, retrying at #{TRY_AGAIN_IN.from_now}"
+          return
+        else
+          SlackNotifierService.call \
+            'Daily CDS file missing, max retry time passed - continuing without todays file'
+        end
```

Comment:
The logic looks perfect here. It would be nice to remove another if level.

A way could be moving the login in a method called "reschedule!", something like this:
 
`reschedule! and return` 

```
def reschedule!
        if still_time_to_reschedule?
          self.class.perform_in(TRY_AGAIN_IN, true)
          logger.info "Daily file missing, retrying at #{TRY_AGAIN_IN.from_now}"

          true # it was rescheduled
        else
          SlackNotifierService.call \
            'Daily CDS file missing, max retry time passed - continuing without todays file'

          false # no reschedule is possible
        end
end
```

Reply:
Good call - now simplified :+1: